packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx

```diff
@@ -78,17 +84,32 @@ function DocSidebarItemCategory({
     if (justBecameActive && collapsed) {
       setCollapsed(false);
     }
+    if (link && !collapsed) {
+      for (const i in items) {
+        if (items.hasOwnProperty(i)) {
+          const childItem = items[i];
+          if (childItem.type === 'link') {
+            if (link.id.includes(childItem.label.toLowerCase())) {
+              setInitialLink(childItem.href);
+            }
+          }
+        }
+      }
+    }
   }, [isActive, wasActive, collapsed]);
 
   const handleItemClick = useCallback(
-    (e) => {
-      e.preventDefault();
-
+    async (e) => {
+      if (!collapsed) {
+        e.preventDefault();
+      }
       if (!menuListHeight) {
         handleMenuListHeight();
       }
-
-      setTimeout(() => setCollapsed((state) => !state), 100);
+      setCollapsed((state) => {
+        return !state;
+      });
+      await sleep(100);
```

Comment:
here you wait 100ms after setting the state, while before we waited before setting the state
that looks different to me